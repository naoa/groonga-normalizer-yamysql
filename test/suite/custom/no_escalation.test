register normalizers/yamysql

table_create text TABLE_NO_KEY
column_create text text COLUMN_SCALAR LongText

tokenize TokenBigram "時代の歌人" NormalizerYaMySQLPartofspeech --mode ADD
tokenize TokenBigram "時代の歌人" NormalizerYaMySQLPartofspeech --mode GET
tokenize TokenBigram "時代の歌人" NormalizerYaMySQL --mode ADD
tokenize TokenBigram "時代の歌人" NormalizerYaMySQL --mode GET

normalize NormalizerYaMySQLPartofspeech "時代の歌人" WITH_TYPES|WITH_CHECKS
normalize NormalizerYaMySQL "時代の歌人" WITH_TYPES|WITH_CHECKS

load --table text
[
["text"],
["時代の歌人"]
]

table_create text-text TABLE_PAT_KEY --key_type ShortText --default_tokenizer TokenBigram --normalizer NormalizerYaMySQLPartofspeech

column_create text-text index COLUMN_INDEX|WITH_POSITION text text

select text --match_columns text --query '時代の歌' --output_columns _key,title,text --limit 1 --match_escalation_threshold -1
