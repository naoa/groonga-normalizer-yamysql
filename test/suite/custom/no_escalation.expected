register normalizers/yamysql
[[0,0.0,0.0],true]
table_create remove_phrases TABLE_PAT_KEY ShortText
[[0,0.0,0.0],true]
load --table remove_phrases
[
{"_key": "の"}
]
[[0,0.0,0.0],1]
table_create text TABLE_NO_KEY
[[0,0.0,0.0],true]
column_create text text COLUMN_SCALAR LongText
[[0,0.0,0.0],true]
tokenize TokenBigram "時代の歌人" NormalizerYaMySQL --mode ADD
[
  [
    0,
    0.0,
    0.0
  ],
  [
    {
      "value": "時代",
      "position": 0,
      "force_prefix": false
    },
    {
      "value": "代歌",
      "position": 1,
      "force_prefix": false
    },
    {
      "value": "歌人",
      "position": 2,
      "force_prefix": false
    },
    {
      "value": "人",
      "position": 3,
      "force_prefix": false
    }
  ]
]
tokenize TokenBigram "時代の歌人" NormalizerYaMySQL --mode GET
[
  [
    0,
    0.0,
    0.0
  ],
  [
    {
      "value": "時代",
      "position": 0,
      "force_prefix": false
    },
    {
      "value": "代歌",
      "position": 1,
      "force_prefix": false
    },
    {
      "value": "歌人",
      "position": 2,
      "force_prefix": false
    }
  ]
]
normalize NormalizerYaMySQL "時代の歌人" WITH_TYPES|WITH_CHECKS
[
  [
    0,
    0.0,
    0.0
  ],
  {
    "normalized": "時代歌人",
    "types": [
      "kanji",
      "kanji",
      "kanji",
      "kanji"
    ],
    "checks": [
      3,
      0,
      0,
      3,
      0,
      0,
      6,
      0,
      0,
      3,
      0,
      0
    ]
  }
]
load --table text
[
["text"],
["時代の歌人"],
["時代 歌人"],
["時代歌人"]
]
[[0,0.0,0.0],3]
table_create text-text TABLE_PAT_KEY --key_type ShortText --default_tokenizer TokenBigram --normalizer NormalizerYaMySQL
[[0,0.0,0.0],true]
column_create text-text index COLUMN_INDEX|WITH_POSITION text text
[[0,0.0,0.0],true]
select text --match_columns text --query '時代の歌' --output_columns _key,title,text --limit -1 --match_escalation_threshold -1
[
  [
    0,
    0.0,
    0.0
  ],
  [
    [
      [
        2
      ],
      [
        [
          "text",
          "LongText"
        ]
      ],
      [
        "時代の歌人"
      ],
      [
        "時代歌人"
      ]
    ]
  ]
]
